### PR TITLE
Feature/toasts

### LINKED ISSUE #232: Feature: Toasts
While table rows are not showing added/removed flash, we should toast when:
- [ ] A book is added via scan input
- [ ] A book is deleted via delete row button
- [ ] Changing the quantity of a volume in a note
- [ ] Creating a new note
- [ ] Deleting or Committing a note

There are a couple options library-wise:
1. [Svelte-toast](https://github.com/zerodevx/svelte-toast): I like it. It's small. But it's also not immediately clear upfront how to plugin to your own toast components. Main reason I'd say we don't go for it is because its currently going through issues with packaging and won't work in ssr. 
2. [Svelte notifications](https://github.com/keenethics/svelte-notifications): Would be my preference. Only thing I can't figure out is how to customise/control positioning more precisely
3. Or, as I did with Edible, we follow [this tutorial](https://dev.to/danawoodman/svelte-quick-tip-creating-a-toast-notification-system-ge3) to add basic functionality

### PR DESCRIPTION
**Pull request by delasous**
Fixes #232 

- First pass at out own Toaster. To be expanded on and open-sourced in time. Covers basic toast rendering/mgmt and ability to pause toasts on hover
- Wires up toasts to basic functionality: create|delete|update note / warehouse

### CODE REVIEW

**Review by silviot — APPROVED** [review]
🖱️🦭: "I clicked around" Seal of Approval

File: pkg/ui/src/Toasts/__tests__/toaster.spec.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		toaster.push(defaultToast);
+
+		const [toast] = get(toaster);
+
+		toaster.pop(toast.id);
+
+		const toasts = get(toaster);
+
+		expect(toasts.length).toBe(0);
+	});
```
Comment (ikusteu) [inline pkg/ui/src/Toasts/__tests__/toaster.spec.ts:44]: I would provide some 'noise' here: add three toasts and remove the middle one (to assert it doesn't take out last/first, but the one with a given id

File: pkg/ui/src/Toasts/toast.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		...toastStore,
+		toast: toastAction
+	};
+};
+
+/**
+ * Creates a toast store to hold and manage toast data
+ */
+export const createToastStore = (toasterStore) => (_toast) => {
+	const toastStore = writable(_toast);
```
Comment (ikusteu) [inline pkg/ui/src/Toasts/toast.ts:32]: I don't see any types here, and am curious as to how TS didn't complain (or am I missing something?)

File: pkg/ui/src/Toasts/__tests__/toaster.spec.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+						toaster.push(defaultToast);
+					}}
+				>
+					<${TestToaster} />
+				</${Fragment}>
+			`
+		);
+		expect(document.querySelector("span")).toHaveTextContent(defaultToast.message);
+	});
```
Comment (ikusteu) [inline pkg/ui/src/Toasts/__tests__/toaster.spec.ts:76]: Is this still necessary. My understanding is that you ditched the context for an imported variable (Map of toasts).

**Review by ikusteu — APPROVED** [review]
Left a couple of comments. AFAIC, this can go in.

File: apps/web-client/src/routes/inventory/outbound/[...id]/+page.svelte [hunk 4]
```diff
@@ -93,16 +102,26 @@
...
+
+	const handleTransactionUpdate = async ({ detail }: CustomEvent<TransactionUpdateDetail>) => {
 		const { matchTxn, updateTxn } = detail;
 		const { isbn, warehouseId = "", quantity = matchTxn.quantity } = updateTxn;
 
-		return note.updateTransaction(matchTxn, { isbn, warehouseId, quantity });
+		await note.updateTransaction(matchTxn, { isbn, warehouseId, quantity });
+
+		// TODO: This doesn't seem to work / get called?
+		toastSuccess(toasts.volumeUpdated(isbn));
```
[the remark this answers is not in the document]
Comment (delasous) [reply apps/web-client/src/routes/inventory/outbound/[...id]/+page.svelte:118]: The entire function is not called

**Pull request merged**